src/frontend/xgboost_json.cc

```diff
@@ -25,45 +43,363 @@ namespace frontend {
+}
+bool GradientBoosterHandler::String(const char *str,
+                                    std::size_t length,
+                                    bool copy) {
+  fmt::string_view name{str, length};
+  if (name != "gbtree") {
+    LOG(ERROR) << "Only GBTree-type boosters are currently supported.";
+    return false;
+  } else {
+    return true;
+  }
+}
+bool GradientBoosterHandler::StartObject() {
+  if (push_key_handler<GBTreeModelHandler, treelite::Model>("model",
+                                                            m_output)) {
+    return true;
+  } else {
+    LOG(ERROR) << "Key \"" << get_cur_key()
+               << "\" not recognized. Is this a GBTree-type booster?";
+    return false;
+  }
+}
+
+/******************************************************************************
+ * ObjectiveHandler
+ * ***************************************************************************/
+bool ObjectiveHandler::StartObject() {
+  return (push_key_handler<IgnoreHandler>("reg_loss_param") ||
+          push_key_handler<IgnoreHandler>("poisson_regression_param") ||
+          push_key_handler<IgnoreHandler>("tweedie_regression_param") ||
+          push_key_handler<IgnoreHandler>("softmax_multiclass_param") ||
+          push_key_handler<IgnoreHandler>("lambda_rank_param") ||
+          push_key_handler<IgnoreHandler>("aft_loss_param"));
+}
+
+bool ObjectiveHandler::String(const char *str, std::size_t length, bool copy) {
+  return assign_value("name", std::string{str, length}, m_output);
+}
+
+/******************************************************************************
+ * LearnerParamHandler
+ * ***************************************************************************/
+bool LearnerParamHandler::String(const char *str,
+                                 std::size_t length,
+                                 bool copy) {
+  return (assign_value("base_score", strtof(str, nullptr),
+                       m_output.param.global_bias) ||
+          assign_value("num_class", static_cast<int>(std::max(atoi(str), 1)),
+                       m_output.num_output_group) ||
+          assign_value("num_feature", atoi(str), m_output.num_feature));
+}
+
+/******************************************************************************
+ * LearnerHandler
+ * ***************************************************************************/
+bool LearnerHandler::StartArray() {
+  return push_key_handler<IgnoreHandler>("eval_metrics"); }
+
+bool LearnerHandler::StartObject() {
+  // "attributes" key is not documented in schema
+  return (push_key_handler<LearnerParamHandler, treelite::Model>(
+              "learner_model_param", m_output) ||
+          push_key_handler<GradientBoosterHandler, treelite::Model>(
+              "gradient_booster", m_output) ||
+          push_key_handler<ObjectiveHandler, std::string>("objective",
+                                                          objective) ||
+          push_key_handler<IgnoreHandler>("attributes"));
```

Comment:
The `attributes` key appears nowhere in the current JSON model schema. What is this line for?

Reply:
It does appear in the JSON output from XGBoost currently. Removing it prevents parsing of current XGBoost JSON models.

Reply:
Oh, and I note that it's not part of the schema at line 290. I'm hoping that's enough to clue future developers in to what's going on there, but we can certainly make that comment louder if necessary.

Reply:
Got it. Let me file a pull request to add `attributes` to the schema.

Reply:
Great! Thanks, @hcho3. Feel free to tag me for review.